sklearn/_build_utils/__init__.py

```diff
@@ -9,11 +9,10 @@
 import sklearn
 import contextlib
 
-from distutils.version import LooseVersion
-
 from .pre_build_helpers import basic_check_build
 from .openmp_helpers import check_openmp_support
 
+from sklearn.utils.fixes import parse_version
```

Comment:
I am not too sure, but I suspect you can not import anything from within sklearn here because this file is used to build sklearn ...

As I may completely be wrong, I would advise you do:
```
pip uninstall scikit-learn -y
pip install --no-build-isolation --editable .
```

and check whether you can reproduce the problem in the CI

Reply:
I reverted this file to not import scikit-learn in my last commit. Hopefully this should get the CI to at least run the tests (i.e. it will avoid that all the CI break during the build).